src/component/addons/AddonList/ConfiguredAddons/ConfiguredAddons.tsx

```diff
@@ -90,36 +90,24 @@ const ConfiguredAddons = ({ getAddonIcon }: IConfigureAddonsProps) => {
                 secondary={addon.description}
             />
             <ListItemSecondaryAction>
-                <ConditionallyRender
-                    condition={hasAccess(UPDATE_ADDON)}
-                    show={
-                        <IconButton
-                            size="small"
-                            title={
-                                addon.enabled ? 'Disable addon' : 'Enable addon'
-                            }
-                            onClick={() => toggleAddon(addon)}
-                        >
-                            <ConditionallyRender
-                                condition={addon.enabled}
-                                show={<Visibility />}
-                                elseShow={<VisibilityOff />}
-                            />
-                        </IconButton>
-                    }
-                />
-                <ConditionallyRender
-                    condition={hasAccess(DELETE_ADDON)}
-                    show={
-                        <IconButton
-                            size="small"
-                            title="Remove addon"
-                            onClick={() => onRemoveAddon(addon)}
-                        >
-                            <Delete />
-                        </IconButton>
-                    }
-                />
+                <PermissionIconButton
+                    permission={UPDATE_ADDON}
+                    tooltip={addon.enabled ? 'Disable addon' : 'Enable addon'}
+                    onClick={() => toggleAddon(addon)}
+                >
+                    <ConditionallyRender
```

Comment:
It would be better to have the conditional render on the outside of the PermissionIconButton